eva/executor/overwrite_executor.py

```diff
@@ -0,0 +1,110 @@
+        table_name = table_info.table_name
+        table_obj = self.catalog.get_table_catalog_entry(table_name, database_name)
+        storage_engine = StorageEngine.factory(table_obj)
+        batch_mem_size = 30000000
+        operation = self.node.operation
+
+        if table_obj.table_type == TableType.VIDEO_DATA:
+            batches = storage_engine.read(table_obj, batch_mem_size)
+
+            # storage_engine.drop(table_obj)
+            # self.catalog.delete_table_catalog_entry(table_obj)
+            #new_table_obj = (
+                #self.catalog.create_and_insert_multimedia_table_catalog_entry(
+                    #table_name, FileFormatType.VIDEO
+                #)
+            #)
+            for batch in batches:
+                video_paths = list(set(batch.frames['name']))
+
+                new_video_paths = []
+                for video_path in video_paths:
+                    cap = cv2.VideoCapture(video_path)
+                    fc = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+                    fw = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+                    fh = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+                    video = np.empty((fc, fh, fw, 3), np.dtype('uint8'))
+
+                    has_next = True
+                    fn = 0
+                    while has_next:
+                        has_next, img = cap.read()
+                        if has_next:
+                            video[fn] = img
+                        fn += 1
+
+                    modified = None
+
+                    # a MUX for operations
+                    if operation == "flip-horizontal":
+                        modified = np.flip(video, axis=2)
+                    elif operation == "flip-vertical":
+                        modified = np.flip(video, axis=1)
+
+                    if modified is not None:
+                        dirs = video_path.split('/')
+                        dirs.append(dirs[-1])
+                        dirs[-2] = 'modified'
+                        modified_dir = '/'.join(dirs[:-1])
+                        if not os.path.exists(modified_dir):
+                            os.mkdir(modified_dir)
+                        new_video_path = '/'.join(dirs)
+                        new_video_paths.append(new_video_path)
+
+                        out = cv2.VideoWriter(new_video_path, cv2.VideoWriter_fourcc(*'mp4v'), fc, (fw, fh))
+                        for fn in range(fc):
+                            out.write(modified[fn])
+                        out.release()
+                    cap.release()
+
+                storage_engine.clear(table_obj, video_paths)
+                storage_engine.write(table_obj, Batch(pd.DataFrame({"file_path": new_video_paths})))
+
+            yield Batch(
+                pd.DataFrame(
+                    {"Table successfully overwritten by: {}".format(operation)},
+                    index=[0],
+                )
+            )
+        else:
```

Comment:
Instead of using a try catch, we are using asserts now. Please look at the insert executor for reference.

Reply:
resolved